src/time_compat.erl

```diff
@@ -43,160 +43,242 @@
+    {monotonic_time, monotonic_time_pre_18, monotonic_time_post_18, 1},
+    {erlang_system_time, erlang_system_time_pre_18, erlang_system_time_post_18, 0},
+    {erlang_system_time, erlang_system_time_pre_18, erlang_system_time_post_18, 1},
+    {os_system_time, os_system_time_pre_18, os_system_time_post_18, 0},
+    {os_system_time, os_system_time_pre_18, os_system_time_post_18, 1},
+    {time_offset, time_offset_pre_18, time_offset_post_18, 0},
+    {time_offset, time_offset_pre_18, time_offset_post_18, 1},
+    {convert_time_unit, convert_time_unit_pre_18, convert_time_unit_post_18, 0},
+    {timestamp, timestamp_pre_18, timestamp_post_18, 0},
+    {unique_integer, unique_integer_pre_18, unique_integer_post_18, 0},
+    {unique_integer, unique_integer_pre_18, unique_integer_post_18, 1}]).
+
 -export([monotonic_time/0,
-	 monotonic_time/1,
-	 erlang_system_time/0,
-	 erlang_system_time/1,
-	 os_system_time/0,
-	 os_system_time/1,
-	 time_offset/0,
-	 time_offset/1,
-	 convert_time_unit/3,
-	 timestamp/0,
-	 unique_integer/0,
-	 unique_integer/1,
-	 monitor/2,
-	 system_info/1,
-	 system_flag/2]).
+         monotonic_time_pre_18/0,
+         monotonic_time_post_18/0,
+         monotonic_time/1,
+         monotonic_time_pre_18/1,
+         monotonic_time_post_18/1,
+         erlang_system_time/0,
+         erlang_system_time_pre_18/0,
+         erlang_system_time_post_18/0,
+         erlang_system_time/1,
+         erlang_system_time_pre_18/1,
+         erlang_system_time_post_18/1,
+         os_system_time/0,
+         os_system_time_pre_18/0,
+         os_system_time_post_18/0,
+         os_system_time/1,
+         os_system_time_pre_18/1,
+         os_system_time_post_18/1,
+         time_offset/0,
+         time_offset_pre_18/0,
+         time_offset_post_18/0,
+         time_offset/1,
+         time_offset_pre_18/1,
+         time_offset_post_18/1,
+         convert_time_unit/3,
+         convert_time_unit_pre_18/3,
+         convert_time_unit_post_18/3,
+         timestamp/0,
+         timestamp_pre_18/0,
+         timestamp_post_18/0,
+         unique_integer/0,
+         unique_integer_pre_18/0,
+         unique_integer_post_18/0,
+         unique_integer/1,
+         unique_integer_pre_18/1,
+         unique_integer_post_18/1,
+         monitor/2,
+         system_info/1,
+         system_flag/2]).
 
 monotonic_time() ->
+    code_version:update(?MODULE),
+    time_compat:monotonic_time().
+
+monotonic_time_post_18() ->
+	erlang:monotonic_time().
+
+monotonic_time_pre_18() ->
+    erlang_system_time_fallback().
+
+monotonic_time(Unit) ->
+    code_version:update(?MODULE),
+    time_compat:monotonic_time(Unit).
+
+monotonic_time_post_18(Unit) ->
     try
-	erlang:monotonic_time()
+        erlang:monotonic_time(Unit)
     catch
-	error:undef ->
-	    %% Use Erlang system time as monotonic time
-	    erlang_system_time_fallback()
+        error:badarg ->
+            erlang:error(badarg, [Unit])
```

Comment:
Why do you catch exceptions from the official API? I believe the compat modules should not change anything, it should be a transparent wrapper of the API.